eng/common/scripts/get-codeowners.ps1

```diff
@@ -13,34 +16,89 @@ if (!(Test-Path $codeOwnersLocation)) {
 }
 
 $codeOwnersContent = Get-Content $codeOwnersLocation
+$previousLine
 
 foreach ($contentLine in $codeOwnersContent) {
   if (-not $contentLine.StartsWith("#") -and $contentLine){
     $splitLine = $contentLine -split "\s+"
     
     # CODEOWNERS file can also have labels present after the owner aliases
     # gh aliases start with @ in codeowners. don't pass on to API calls
-    $ownedFolders[$splitLine[0].ToLower().Trim("/")] = ($splitLine[1..$($splitLine.Length)] `
-      | ? { $_.StartsWith("@") } `
-      | % { return $_.substring(1) }) -join ","
+
+    $aliases = ($splitLine[1..$($splitLine.Length)] | ? { $_.StartsWith("@") } | % { return $_.substring(1) }) -join ","
+    $labels = ""
+
+    if ($null -ne $previousLine -and $previousLine.StartsWith("# PRLabel:"))
+    {
+      $previousLine = $previousLine.Replace("# PRLabel: ","")
+      $splitPrevLine = $previousLine -split "%" 
+      $labels = ($splitPrevLine[1..$($splitPrevLine.Length)] | % { return $_.Trim() }) -join ","
+    }
+
+    $ownedFolders[$splitLine[0].ToLower().Trim("/")] = @{ Aliases = $aliases; Labels = $labels }
   }
+  $previousLine = $contentLine
 }
 
 $results = $ownedFolders[$target]
 
 if ($results) {
-  Write-Host "Found a folder $results to match $target"
-  
-  if ($VsoVariable) {
-    $alreadyPresent = [System.Environment]::GetEnvironmentVariable($VsoVariable)
+  Write-Host "Found a folder to match $target"
+  $aliases = $results.Aliases -split ","
+  $users
+  $teams
+
+  if ($AuthToken) {
+    $headers = @{
+      Authorization = "bearer $AuthToken"
```

Comment:
Do we need an auth token to query the users?

Reply:
Not necessary but using authToken increases the rate limit.